packages/mail-client/src/mail-client.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  Destination,
+  DestinationOrFetchOptions,
+  resolveDestinationWithType,
+  toDestinationNameUrl
+} from '@sap-cloud-sdk/connectivity';
+import { createLogger } from '@sap-cloud-sdk/util';
+import nodemailer, { SentMessageInfo, Transporter } from 'nodemailer';
+import { MailOptions, MailResponse } from './mail-client-types';
+
+const logger = createLogger({
+  package: 'mail-client',
+  messageContext: 'mail-client'
+});
+
+function getMailDestinationOriginalProperties(
+  destination: Destination
+): Record<string, any> {
+  const originalProperties = destination.originalProperties;
+  if (!originalProperties) {
+    throw Error(
+      `The destination '${toDestinationNameUrl(
+        destination
+      )}' does not contain property configurations.`
+    );
+  }
+  return originalProperties;
+}
+
+function getHostAndPort(
+  destination: Destination,
+  originalProperties: Record<string, any>
+): { host: string; port: number } {
+  const host = originalProperties['mail.smtp.host'];
+  const port = originalProperties['mail.smtp.port'];
+
+  if (!(host && port)) {
+    throw Error(
+      `The destination '${toDestinationNameUrl(
+        destination
+      )}' does not contain host or port information as properties.`
+    );
+  }
+  return { host, port: parseInt(port) };
+}
+
+function getCredentials(
+  destination: Destination,
+  originalProperties: Record<string, any>
+): { user: string; password: string } {
+  const user = originalProperties['mail.user'];
+  const password = originalProperties['mail.password'];
+
+  if (!(user && password)) {
+    throw Error(
+      `The destination '${toDestinationNameUrl(
+        destination
+      )}' does not contain user or password information as properties.`
+    );
+  }
+  return { user, password };
+}
+
+function createTransport(
+  destination: Destination,
+  originalProperties: Record<string, any>
+): Transporter<SentMessageInfo> {
+  const { host, port } = getHostAndPort(destination, originalProperties);
+  const { user, password } = getCredentials(destination, originalProperties);
+
+  return nodemailer.createTransport({
+    host,
+    port,
+    // Defines if the connection should use SSL (if true) or not (if false).
+    secure: false,
+    auth: {
+      user,
+      pass: password
+    },
+    tls: {
+      /**
+       * If true the server will reject any connection which is not
+       * authorized with the list of supplied CAs. This option only has an
+       * effect if requestCert is true.
+       */
+      /** Disable tls config to fix the self signed certificate error. */
+      rejectUnauthorized: false
```

Comment:
[req] We should not do this. If I get it right we should people to configure this because this could open up for man in the middle attacks. In the repo they always recomend to use the `rejectUnauthorized` but I think this is not good: https://github.com/nodemailer/nodemailer/issues?q=self+signed. If you have a test could you add the `ca:...` property to the tls with the buffer as shown here: https://nodejs.org/api/tls.html#tlsconnectoptions-callback. If this works we should read the certificates from the cloud platform trust store (see here https://github.com/SAP/cloud-sdk-js/blob/50b9142fc72a67a68816c030c1490773d3c6d2a7/packages/connectivity/src/scp-cf/destination/destination-service.ts#L182) and put it because this is the secure way to do it.

Reply:
We should allow to pass it as an option to switch it off for some intranet/test case.

Reply:
As discussed, we can work on a separate PR for the certificates.

Notes:
- we did not find the same way like http destination, where you can upload a certificate, meaning it should be passed as a parameter
- the `nodemailer` accepts the following options, which is actually the `SecureContextOptions` of `node`

``` ts
tls: {
      cert: xxx,
      ca: xxx,
      rejectUnauthorized: xxx
    }
```